design/src/rfcs/rfc0007_split_release_process.md

```diff
@@ -0,0 +1,143 @@
+The `next` branch's CI will behave similar to the previous CI, except that it runs an additional
+step beforehand to ensure the correct Smithy runtime crates are in place. The process as a whole
+looks as follows:
+
+1. Make a shallow clone of `aws-sdk-rust/next`
+2. Retrieve the `smithy-rs` commit hash that was used to generate the SDK from a file
+   that was generated alongside the rest of the build artifacts from `smithy-rs` and
+   copied into `aws-sdk-rust`.
+3. Make a shallow clone of `smithy-rs` at the correct commit hash
+4. Use a script to add a `[patch]` section to all the AWS SDK crates to point to the
+   Smithy runtime crates from the local clone of `smithy-rs`.
+   For example:
+```toml
+# The dependencies section is left alone, but is here for context
+[dependencies]
+# Some version of aws-smithy-types that isn't on crates.io yet, referred to as `<unreleased>` below
+aws-smithy-types = "<unreleased>"
+
+# This patch section gets added by the script
+[patch.crates-io]
+aws-smithy-types = { version = "<unreleased>", path = "path/to/local/smithy-rs/rust-runtime/aws-smithy-types"}
+```
+5. Run CI as normal.
+
+When it's time to cut a release, the releaser will do the following:
+
+1. Tag `smithy-rs` with the desired version number
+2. Wait for CI to build artifacts for the tagged release
+3. Pull-request the SDK artifacts over to `aws-sdk-rust/next`
+4. Wait for successful CI in `aws-sdk-rust/next`
+5. Download the Smithy runtime crates build artifact and publish it to crates.io
+6. Pull-request merge `aws-sdk-rust/next` into `aws-sdk-rust/main`
+7. Wait for successful CI in `main` (this time actually running against the crates.io Smithy runtime crates)
+8. Tag release for `main`
+9. Publish SDK with publisher tool
+
+**Note:** `smithy-rs` will need to do the same patching in CI as `aws-sdk-rust/next` since the generated
+SDK will not have path dependencies for the Smithy runtime crates (since they are a publish-ready bundle
+intended for landing in `aws-sdk-rust`). The script that does this patching can live in `smithy-rs` and be
+reused by `aws-sdk-rust`.
+
+Alternatives Considered
+-----------------------
+
+### Snapshotting the Smithy runtime crates in `aws-sdk-rust`
+
+We could keep a copy of the Smithy runtime crates used by the SDK in `aws-sdk-rust`.
+This would look mostly the same as the proposed solution, except that rather than cloning
+`smithy-rs` in CI, we would use the local snapshot. The CI for the `main` branch CI would
+need to modify the crate manifests so that the Smithy runtime crates published to
+crates.io are used instead of the local copies.
+
+This has the advantage that a customer having an issue with the current release could get
+a fix sooner by patching their own project's crate manifest to use the `aws-sdk-rust/next`
+branch before a release is cut. However, it also has the disadvantage that the changes
+CI has to make while verifying the `main` branch are more intrusive.
```

Comment:
We should discuss this alternative vs. the proposed solution in depth. I think the alternative might actually be more compelling, and the intrusiveness of the manifest changes is pretty minor.

Reply:
My main question about the suggested vs alternative approach is what do we gain by not having a local copy of the runtime crates inside `aws-sdk-rust`? Separation of concerns?

I am asking this because the alternative approach seems to be involving less "machinery" in the release process. It is however true that the CI changes will be more intrusive..

Reply:
After thinking about this more, I'm preferring the alternate approach to the proposed approach. I will swap the two in the next revision.